neqo-transport/src/connection.rs

```diff
@@ -1383,13 +1397,14 @@ impl Connection {
     }
 
     /// Close the connection.
-    pub fn close(&mut self, now: Instant, error: AppError, msg: &str) {
-        self.set_state(State::Closing {
-            error: ConnectionError::Application(error),
-            frame_type: 0,
-            msg: msg.into(),
-            timeout: self.get_closing_period_time(now),
-        });
+    pub fn close<S>(&mut self, now: Instant, app_error: AppError, msg: S)
+    where
+        S: AsRef<str>,
```

Comment:
you like this better than
```
    pub fn close(
        &mut self,
        error: ConnectionError,
        frame_type: FrameType,
        message: impl AsRef<str>,
```
I like `impl Trait` in argument position but I know opinions differ.

Reply:
I don't know why I keep forgetting that.  I also prefer that.